src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
I think you could use [`async.race`](http://caolan.github.io/async/docs.html#race) instead, then you don't have to do the manual success tracking